src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -178,6 +182,8 @@ impl ThinPool {
         let spare_segments = segments_list.pop().expect("len(segments_list) == 2");
         let meta_segments = segments_list.pop().expect("len(segments_list) == 1");
 
+        let backstore_device = backstore.device();
```

Comment:
Here we get the backstore device just once, instead of calling the method every time, to make it obvious that it is expected to be constant throughtout the method.